packages/flutter_tools/lib/src/asset.dart

```diff
@@ -15,8 +16,48 @@ import 'devfs.dart';
 import 'flutter_manifest.dart';
 import 'globals.dart';
 
+const AssetBundleFactory _kManifestFactory = const _ManifestAssetBundleFactory();
+
+/// Injected factory class for spawning [AssetBundle] instances.
+abstract class AssetBundleFactory {
+  /// The singleton instance, pulled from the [AppContext].
+  static AssetBundleFactory get instance => context == null
+      ? _kManifestFactory
+      : context.putIfAbsent(AssetBundleFactory, () => _kManifestFactory);
+
+  /// Creates a new [AssetBundle].
+  AssetBundle createBundle();
+}
+
+abstract class AssetBundle {
+  factory AssetBundle.fixed(String projectRoot, String projectAssets) =>
+      new _ManifestAssetBundle.fixed(projectRoot, projectAssets);
+
+  Map<String, DevFSContent> get entries;
+
+  bool needsBuild({String manifestPath: _ManifestAssetBundle.defaultManifestPath});
+
+  /// Returns 0 for success; non-zero for failure.
+  Future<int> build({
+    String manifestPath: _ManifestAssetBundle.defaultManifestPath,
+    String workingDirPath,
+    String packagesPath,
+    bool includeDefaultFonts: true,
+    bool reportLicensedPackages: false
+  });
+}
+
+class _ManifestAssetBundleFactory implements AssetBundleFactory {
+  const _ManifestAssetBundleFactory();
+
+  @override
+  AssetBundle createBundle() => new _ManifestAssetBundle();
+}
+
+
 /// A bundle of assets.
```

Comment:
Should this comment move to the abstract class? (though it doesn't look _that_ useful)

Reply:
I just removed it since it breaks https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-useless-documentation